app/services/foreman/plugin.rb

```diff
@@ -254,8 +255,8 @@ def allowed_template_variables(*variables)
 
     # Add Compute resource
     def compute_resource(provider)
-      SETTINGS[provider.name.split('::').last.downcase.to_sym] = true
-      ComputeResource.register_provider provider
+      return if @compute_resources.include?(provider)
+      @compute_resources << provider
```

Comment:
Good point, since that's stored indefinitely.  Plugin storage now returns the class name and checks that the registration passed in a class before attempting to call .name on it.